lib/wanda/execution.ex

```diff
@@ -27,21 +27,43 @@ defmodule Wanda.Execution do
   defp maybe_start_execution(_, _, _, [], _, _), do: {:error, :no_checks_selected}
 
   defp maybe_start_execution(execution_id, group_id, targets, checks, env, config) do
-    case DynamicSupervisor.start_child(
-           Supervisor,
-           {Server,
-            execution_id: execution_id,
-            group_id: group_id,
-            targets: targets,
-            checks: checks,
-            env: env,
-            config: config}
-         ) do
-      {:ok, _} ->
-        :ok
+    with false <- execution_for_group_id_already_running(group_id),
+         {:ok, _} <-
+           DynamicSupervisor.start_child(
+             Supervisor,
+             {Server,
+              execution_id: execution_id,
+              group_id: group_id,
+              targets: targets,
+              checks: checks,
+              env: env,
+              config: config}
+           ) do
+      :ok
+    else
+      true ->
+        {:error, :group_already_running}
 
       {:error, _} = error ->
         error
     end
   end
+
+  defp execution_for_group_id_already_running(group_id) do
+    :global.registered_names()
+    |> Enum.flat_map(fn entry ->
+      case entry do
+        {Server, execution_id} -> [{Server, execution_id}]
+        _ -> []
+      end
+    end)
+    |> Enum.any?(fn name ->
+      %{group_id: g} =
+        name
+        |> :global.whereis_name()
+        |> :sys.get_state()
```

Comment:
We should never introspect the state  of an actor: https://www.erlang.org/doc/man/sys.html#get_state-2

See the green note, this function is there just for debugging purpose

Reply:
Yes, I had already changed this locally